lib/Graph/Nodes.cpp

```diff
@@ -636,6 +648,8 @@ bool TanhGradNode::verify() const {
   return isValid;
 }
 
+bool ExpNode::verify() const { return verifyExp(getInput(), getResult()); }
```

Comment:
nit: No need to use a helper function `verifyExp()` here, since it's only being used by this one Node.

Reply:
Intention was to future proof this for Grad implementation, as was being done by the other nodes of similar kind. As a policy do you encourage future proofing?

Reply:
If all you intend to add grad support in the near future then it's fine to keep, NBD. Otherwise it's not very hard to modify in the future if/when we add grad support, and so I'd say to keep it inline since it only adds a level of indirection without benefit.

Reply:
OK, makes sense.